workers/api-server/src/router.rs

```diff
@@ -130,19 +130,29 @@ impl<
 
         // Forward to the typed handler
         let res = self.handle_typed(headers, req_body, url_params).await;
-        if let Ok(resp) = res {
-            // Serialize the response into a body. We explicitly allow for all cross-origin
-            // requests, as users connecting to a locally-run node have a different origin
-            // port.
-
-            // TODO: Either remove this in the future, or ensure that no sensitive
-            // information can leak from cross-origin requests.
-            Response::builder()
+        match res {
```

Comment:
I'd propose simplify this code by setting up all common fields on the response outside of this match, and then inside the match we apply a body and status if necessary:
```rust
let mut builder = Response::builder().header("Access-Control-Allow-Origin", "*");
match res {
    Ok(resp) => builder.body(...),
    Err(ApiServerError::HttpStatusCode(status, msg)) => builder.status(status).body(Body::from(msg)),
    Err(_) => builder.status(...),
}.unwrap()
```